relay-server/src/actors/outcome.rs

```diff
@@ -475,24 +479,40 @@ pub enum OutcomeError {
 struct HttpOutcomeProducer {
     config: Arc<Config>,
     unsent_outcomes: Vec<TrackRawOutcome>,
-    pending_flush_handle: Option<SpawnHandle>,
+    // Sleep must be `Box<Pin>` in order for it to be Unpin.
+    // See: https://docs.rs/tokio/1.21.0/tokio/time/struct.Sleep.html#examples
+    pending_flush_handle: Either<future::Pending<()>, Pin<Box<tokio::time::Sleep>>>,
 }
 
 impl HttpOutcomeProducer {
     pub fn create(config: Arc<Config>) -> Result<Self, ServerError> {
         Ok(Self {
             config,
             unsent_outcomes: Vec::new(),
-            pending_flush_handle: None,
+            pending_flush_handle: Either::Left(future::pending()),
         })
     }
+
+    pub fn start(mut self) -> ServiceAddr<Self> {
+        let (tx, mut rx) = mpsc::unbounded_channel::<HttpOutcomeProducerMessages>();
+
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
+                    Some(message) = rx.recv() => self.handle_message(message),
+                    () = &mut self.pending_flush_handle => self.send_batch(),
+                    else => break,
```

Comment:
IIUC this else branch will execute when the `rx.recv()` returns `None`, i.e. all the `Addr`s are closed (i.e. the registry is dropped).  So this is kind of the implicit stopping of this actor once everything else is shut down?

Reply:
So far, this is the model all our services have followed and is the same behavior as actix has had. +1 on investigating better APIs in a follow-up.

Reply:
I was surprised this wasn't something we wanted to handle with our shutdown signalling.  But no need to change that now as it would be a behaviour change.  Just wanted to double-check this wasn't a behaviour change.